cargo-dist/src/lib.rs

```diff
@@ -609,11 +594,19 @@ fn do_generate_preflight_checks(dist: &DistGraph) -> Result<()> {
 /// Generate any scripts which are relevant (impl of `cargo dist generate`)
 pub fn do_generate(cfg: &Config, args: &GenerateArgs) -> Result<()> {
     let dist = gather_work(cfg)?;
-    do_generate_preflight_checks(&dist)?;
+
+    run_generate(&dist, &args.modes, args.check)?;
+
+    Ok(())
+}
+
+/// The inner impl of do_generate
+pub fn run_generate(dist: &DistGraph, modes: &[GenerateMode], check: bool) -> Result<()> {
```

Comment:
Don't know how I feel about the last arg just being a boolean - the other calls to `run_generate()` aren't really clear. I tend to find positional booleans illegible.

Reply:
you're not wrong, i should add an enum, then?